sound/soc/sof/intel/hda-pcm.c

```diff
@@ -136,6 +137,38 @@ int hda_dsp_pcm_trigger(struct snd_sof_dev *sdev,
 	return hda_dsp_stream_trigger(sdev, stream, cmd);
 }
 
+snd_pcm_uframes_t hda_dsp_pcm_pointer(struct snd_sof_dev *sdev,
+				      struct snd_pcm_substream *substream)
+{
+	struct hdac_stream *hstream = substream->runtime->private_data;
+	struct hdac_ext_stream *stream = stream_to_hdac_ext_stream(hstream);
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	snd_pcm_uframes_t pos = 0;
+
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
+		pos = snd_sof_dsp_read(sdev, HDA_DSP_HDA_BAR,
+				       AZX_REG_VS_SDXDPIB_XBASE +
+				       (AZX_REG_VS_SDXDPIB_XINTERVAL *
+					hstream->index));
+	} else {
+		usleep_range(20, 30);
+		snd_sof_dsp_read(sdev, HDA_DSP_HDA_BAR,
+				 AZX_REG_VS_SDXDPIB_XBASE +
+				 (AZX_REG_VS_SDXDPIB_XINTERVAL *
+				  hstream->index));
+		pos = snd_hdac_stream_get_pos_posbuf(hstream);
+	}
+
+	if (pos >= hstream->bufsize)
+		pos = 0;
+
+	pos = bytes_to_frames(substream->runtime, pos);
+
+	dev_dbg(sdev->dev, "PCM: stream %d dir %d position %lu\n",
```

Comment:
dev_vdbg since this will be high frequency ?

Reply:
good, let me change.